packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -1601,6 +1603,11 @@ export class ComboBox extends React.Component<IComboBoxProps, IComboBoxState> {
           isOpen: false,
         });
       }
+
+      // Continue processing the click only after
+      // performing menu close / control focus(inner working)
+      onItemClick && onItemClick(ev, item, index);
```

Comment:
The only things changed as part of this PR is moving lines 1593 and 1594 to here (lines 1609 and 1610)